dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/tools/BeanMethodHandler.java

```diff
@@ -69,7 +73,8 @@ public static record MethodMetadata(String name,
                                         boolean isStructuredContent,
                                         List<Class<? extends Throwable>> businessExceptions,
                                         List<SpecialArgumentMetadata> specialArguments,
-                                        String[] argNames) {}
+                                        String[] argNames,
+                                        Map<TypeVariable<?>, Type> genericMap) {}
```

Comment:
If we put the generic map in the method metadata, we must make sure it's immutable.